### PR TITLE
Add Azure Swarm Master Port endpoint configuration when using docker …

### PR DESCRIPTION
**Pull request by agup006**
…machine. When using docker-machine create with Azure driver, if using --swarm-master, Azure Ubuntu VM automatically opens port 3376. Resolving Issue 640

Signed-off-by: Anurag Gupta <agup006@gmail.com>

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Ooh, very interesting. I haven't tried this but the code looks good to me.

@chanezon @ehazlett Can you please take a look?

**Comment by ehazlett** [comment]
Code looks good.  I still need to give it a run.

File: drivers/azure/azure.go [hunk 1]
```diff
@@ -386,6 +393,15 @@ func (d *Driver) addDockerEndpoint(vmConfig *vmClient.Role) error {
 			Protocol:  "tcp",
 			Port:      d.DockerPort,
 			LocalPort: d.DockerPort}
+		if d.SwarmMaster == true {
```
Comment (nathanleclaire) [inline drivers/azure/azure.go]: This can be abbreviated to just `if d.SwarmMaster {`

File: drivers/azure/azure.go [hunk 2]
```diff
@@ -386,6 +393,15 @@ func (d *Driver) addDockerEndpoint(vmConfig *vmClient.Role) error {
...
+			swarm_ep := vmClient.InputEndpoint{
+				Name:      "docker swarm",
+				Protocol:  "tcp",
+				Port:      d.DockerSwarmMasterPort,
+				LocalPort: d.DockerSwarmMasterPort}
```
Comment (nathanleclaire) [inline drivers/azure/azure.go]: Nit: please put the `}` on the next line instead of the same line here.

**Comment by chanezon** [comment]
It works! Thank you, that fixes this infamous bug!
docker info
Containers: 3
Images: 2
Role: primary
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 2
 pat-swarm-0809-master: pat-swarm-0809-master.cloudapp.net:2376
  └ Containers: 2
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.721 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-61-generic, operatingsystem=Ubuntu 14.04.1 LTS, provider=azure, storagedriver=aufs
 pat-swarm-0809-node-00: pat-swarm-0809-node-00.cloudapp.net:2376
  └ Containers: 1
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.721 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-36-generic, operatingsystem=Ubuntu 14.04.1 LTS, provider=azure, storagedriver=aufs
CPUs: 2
Total Memory: 3.442 GiB

**Comment by nathanleclaire** [comment]
LGTM

**Comment by abronan** [comment]
LGTM thanks @agup006!

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**